main.go

```diff
@@ -102,8 +126,30 @@ func (c *customDNSProviderSolver) Present(ch *v1alpha1.ChallengeRequest) error {
 // value provided on the ChallengeRequest should be cleaned up.
 // This is in order to facilitate multiple DNS validations for the same domain
 // concurrently.
-func (c *customDNSProviderSolver) CleanUp(ch *v1alpha1.ChallengeRequest) error {
-	// TODO: add code that deletes a record from the DNS provider's console
+func (c *ns1DNSProviderSolver) CleanUp(ch *v1alpha1.ChallengeRequest) error {
+	cfg, err := loadConfig(ch.Config)
+	if err != nil {
+		return err
+	}
+
+	zone, domain, err := c.parseChallenge(ch)
+	if err != nil {
+		return err
+	}
+
+	if c.ns1Client == nil {
+		if err := c.setNS1Client(ch, cfg); err != nil {
+			return err
+		}
+	}
+
+	// Create the TXT Record we created in Present
```

Comment:
Typo?  Think `Create` should be `Delete`?

Reply:
🙇